models/intermediate/int_shopify__discount_code_enriched.sql

```diff
@@ -0,0 +1,170 @@
+        codes_precision,
+        combines_with_order_discounts,
+        combines_with_product_discounts,
+        combines_with_shipping_discounts,
+        created_at,
+        customer_selection_all_customers,
+        ends_at, 
+        starts_at,
+        status,
+        title,
+        total_sales_amount,
+        total_sales_currency_code,
+        updated_at,
+        usage_limit,
+        source_relation
+    from {{ var('shopify_discount_code_basic') }}
+),
+
+discount_code_bxgy as (
+
+    select
+        discount_code_id,
+        'bxgy' as discount_type, 
+        applies_once_per_customer,
+        usage_count,
+        codes_count,
+        codes_precision,
+        combines_with_order_discounts,
+        combines_with_product_discounts,
+        combines_with_shipping_discounts,
+        created_at,
+        customer_selection_all_customers,
+        ends_at,
+        starts_at,
+        status,
+        title,
+        total_sales_amount,
+        total_sales_currency_code,
+        updated_at,
+        usage_limit,
+        source_relation
+    from {{ var('shopify_discount_code_bxgy') }}
+),
+
+discount_code_free_shipping as (
+
+    select
+        discount_code_id,
+        'free_shipping' as discount_type, 
+        applies_once_per_customer,
+        usage_count,
+        codes_count,
+        codes_precision,
+        combines_with_order_discounts,
+        combines_with_product_discounts,
+        combines_with_shipping_discounts,
+        created_at,
+        customer_selection_all_customers,
+        ends_at, 
+        starts_at,
+        status,
+        title,
+        total_sales_amount,
+        total_sales_currency_code,
+        updated_at,
+        usage_limit,
+        source_relation
+    from {{ var('shopify_discount_code_free_shipping') }}
+),
+
+{% if var('shopify_using_discount_code_app', False) %}
+
+discount_code_app as (
+
+    select
+        discount_code_id,
+        'app' as discount_type,
+        applies_once_per_customer,
+        usage_count,
+        codes_count,
+        codes_precision,
+        combines_with_order_discounts,
+        combines_with_product_discounts,
+        combines_with_shipping_discounts,
+        created_at,
+        customer_selection_all_customers,
+        ends_at,
+        starts_at,
+        status,
+        title,
+        total_sales_amount,
+        total_sales_currency_code,
+        updated_at,
+        usage_limit,
+        source_relation
+    from {{ var('shopify_discount_code_app') }}
+),
+{% endif %}
+
+discount_applications as (
+
+    select *
+    from {{ var('shopify_discount_application') }}
+),
+
+unified_discount_codes as (
+
+    select * 
+    from discount_code_basic
+    
+    union all
+    
+    select * 
+    from discount_code_bxgy
+
+    union all
+    
+    select * 
+    from discount_code_free_shipping
+
+    {% if var('shopify_using_discount_code_app', False) %}
+    union all
+    select * from discount_code_app
+    {% endif %}
+),
```

Comment:
Is this necessary? Could we instead turn the above ctes into a this one and then apply the unions in the same way? This way we save on repeated steps.

Reply:
Consolidated.